datafusion/core/src/physical_planner.rs

```diff
@@ -670,6 +670,10 @@ impl DefaultPhysicalPlanner {
                 let input_exec = children.one()?;
                 let physical_input_schema = input_exec.schema();
                 let logical_input_schema = input.as_ref().schema();
+                let physical_input_schema_from_logical: Arc<Schema> =
+                    logical_input_schema.as_ref().clone().into();
+
+                debug_assert_eq!(physical_input_schema_from_logical, physical_input_schema, "Physical input schema should be the same as the one converted from logical input schema. Please file an issue or send the PR");
```

Comment:
The main goal of the change is to ensure they are the same.

Reply:
Nice!

Did you consider making this function return an `internal_error` rather than debug_assert ? 

If we are concerned about breaking existing tests, we could add a config setting like `datafusion.optimizer.skip_failed_rules` to let users bypass the check

Reply:
The change here is to ensure they are equivalent, thus I don't think we should let user bypass the check 🤔